packages/@aws-cdk/assertions/README.md

```diff
@@ -53,6 +53,14 @@ assert.templateMatches({
 });
 ```
 
+The `asObject()` can also be used to perform [snapshot
+testing](https://jestjs.io/docs/snapshot-testing).
+
+```ts
+// using jest
+expect(Template.fromStack(stack).asObject()).toMatchSnapshot();
```

Comment:
Feels like this should be called `toJSON()` and then there is a slight chance this will even work without calling toJSON() because JSON.stringify respects it

Reply:
But this isn't JSON. It could very well be parsed YAML, no?
It's basically an "in memory" template, so "object" felt suitable.

Reply:
[`toJSON()`] in JavaScript is the standard name for a method that converts an object to a JSON-serializable representation, which I think is exactly what you basically want here "template from stack to JSON".

[`toJSON()`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#tojson_behavior

Reply:
Ok, fair enough. I will change.

Do you feel the same about `fromObject()`? It doesn't have the same nice-ties and my point about it being from YAML rings more true.

Reply:
I think it should still be `fromJSON()` so that there is a semantic relationship between the two methods.

```ts
Template.fromJSON();
myTemplate.toJSON()
```

Reply:
The fromJSON method does not accept a JSON string, it accepts a JavaScript Object, so it won't work for YAML. Right?

I think `Template.fromString()` should accept YAML (which is a superset of JSON), so it means users should be able to pass in either YAML or JSON to this method and it should work. right?

Reply:
> it won't work for YAML. Right?

Running a YAML file through a generic YAML deserializer and using the result in this method will work. So customers can choose to use their own deserializer

I think I can get behind the general notion that `fromString()` should accept JSON or YAML string. And we can then change this to `fromJSON()`.

Reply:
> Running a YAML file through a generic YAML deserializer 

...means that it's no longer YAML :-)

---

Don't get me wrong - I understand why JSON might not be an ideal name, but it's a good enough name I think